src/lib/chain/offer/updateOffers.ts

```diff
@@ -0,0 +1,456 @@
+
+import { commandObj } from "../../commandObj.js";
+import { CLI_NAME } from "../../const.js";
+import { getDealClient, signBatch, populateTx } from "../../dealClient.js";
+import { uint8ArrayToHex } from "../../helpers/typesafeStringify.js";
+import { splitErrorsAndResults } from "../../helpers/utils.js";
+import { confirm } from "../../prompt.js";
+import {
+  cidStringToCIDV1Struct,
+  peerIdHexStringToBase58String,
+  cidHexStringToBase32,
+} from "../conversions.js";
+import { ptFormatWithSymbol } from "../currencies.js";
+import { assertProviderIsRegistered } from "../providerInfo.js";
+
+import {
+  type OffersArgs,
+  resolveOffersFromProviderConfig,
+  type EnsureOfferConfig,
+} from "./offer.js";
+
+export async function updateOffers(flags: OffersArgs) {
+  const offers = await resolveOffersFromProviderConfig(flags);
+  const offersFoundOnChain = filterOffersFoundOnChain(offers);
+  const populatedTxs = await populateUpdateOffersTxs(offersFoundOnChain);
+
+  const txsToSign = [
+    populatedTxs.flatMap(({ txsToRunBeforeOthers }) => {
+      return txsToRunBeforeOthers.map(({ tx }) => {
+        return tx;
+      });
+    }),
+    populatedTxs.flatMap(({ txs }) => {
+      return txs.map(({ tx }) => {
+        return tx;
+      });
+    }),
+  ].flat();
+
+  if (txsToSign.length === 0) {
+    commandObj.logToStderr("No changes found for selected offers");
+    return;
+  }
+
+  printOffersToUpdateInfo(populatedTxs);
+
+  if (
+    !(await confirm({
+      message: "Would you like to continue",
+      default: true,
+    }))
+  ) {
+    commandObj.logToStderr("Offers update canceled");
+    return;
+  }
+
+  await assertProviderIsRegistered();
+  await signBatch(txsToSign);
+}
+
+type OnChainOffer = Awaited<
+  ReturnType<typeof filterOffersFoundOnChain>
+>[number];
+
+function filterOffersFoundOnChain(offers: EnsureOfferConfig[]) {
+  const [offersNotFoundOnChain, offersToUpdateFoundOnChain] =
+    splitErrorsAndResults(
+      offers,
+      ({
+        offerInfo: { offerInfo, offerIndexerInfo } = {
+          offerInfo: undefined,
+          offerIndexerInfo: undefined,
+        },
+        offerId,
+        ...rest
+      }) => {
+        return offerId === undefined ||
+          offerInfo === undefined ||
+          offerIndexerInfo === undefined
+          ? { error: { ...rest, offerId } }
+          : { result: { ...rest, offerInfo, offerIndexerInfo, offerId } };
+      },
+    );
+
+  if (offersNotFoundOnChain.length > 0) {
+    commandObj.warn(
+      `Some of the offers are not found on chain:\n${offersNotFoundOnChain
+        .map(({ offerName, offerId }) => {
+          return `${offerName}${offerId === undefined ? "" : ` (${offerId})`}`;
+        })
+        .join(
+          "\n",
+        )}\n\nPlease make sure the offers exist. If the offers don't exist you can create them using '${CLI_NAME} provider offer-create' command`,
+    );
+  }
+
+  return offersToUpdateFoundOnChain;
+}
+
+type PeersOnChain = {
```

Comment:
I'd move all types to the top of the file, so it's easy to see the whole nomenclature / domain before reading the operations applied.

Reply:
can do